src/App.tsx

```diff
@@ -1,24 +1,181 @@
+      title,
+      completed: false,
+    };
+
+    setTempTodo(newTodo);
+    setIsDisabled(true);
+    setError('');
+
+    fetchAddTodo(USER_ID, newTodo)
+      .then((res) => {
+        setTodos((prevTodo) => {
+          return [...prevTodo, res];
+        });
+
+        setFilter(Filter.ALL);
+      })
+      .catch(() => {
+        setError('Unable to add a todo');
+
+        const timer = setTimeout(() => {
+          setError('');
+        }, 3000);
+
+        return () => {
+          clearTimeout(timer);
+        };
+      })
+      .finally(() => {
+        setIsDisabled(false);
+        setTempTodo(null);
+      });
+  };
+
+  const deleteTodo = (id: number) => {
+    setDeleteTodoId(id);
+    remove(id)
+      .then(() => {
+        // const result = todos.filter(todo => todo.id !== id);
+
+        setTodos((prevTodo) => {
+          return prevTodo.filter(todo => todo.id !== id);
+        });
+
+        // setTodos(result);
+        setError('');
+      })
+      .catch(() => {
+        setError('Unable to delete a todo');
+
+        const timer = setTimeout(() => {
+          setError('');
+        }, 3000);
+
+        return () => {
+          clearTimeout(timer);
+        };
+      })
+      .finally(() => {
+        setDeleteTodoId(0);
+      });
+  };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const arrTodos = await fetchTodos(USER_ID.toString());
+
+        setTodos(arrTodos);
+      } catch {
+        setError('Unable to fetch todos');
+      }
+    };
+
+    if (USER_ID) {
+      fetchData();
+    }
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filter) {
+        case Filter.ACTIVE:
+          return !todo.completed;
+        case Filter.COMPLETED:
+          return todo.completed;
+        case Filter.ALL:
+          return true;
+        default:
+          return todo;
+      }
+    });
+  }, [filter, todos]);
+
+  const handleClearCompleted = () => {
+
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+
+        <Header
+          todos={todos}
+          addTodo={addTodo}
+          newTodoTitle={newTodoTitle}
+          setNewTodoTitle={setNewTodoTitle}
+          isDisabled={isDisabled}
+        />
+
+        <TodoList
+          todos={filteredTodos}
+          tempTodo={tempTodo}
+          deleteTodo={deleteTodo}
+          deleteTodoId={deleteTodoId}
+        />
+
+        {!!todos.length && (
```

Comment:
don't need to use '!!' here, because the '&&' will already transform it to the boolean